app/services/encrypted_attribute.rb

```diff
@@ -0,0 +1,76 @@
+class EncryptedAttribute
+  attr_accessor :user_access_key
+  attr_reader :encrypted, :decrypted
+
+  def self.new_user_access_key(cost: nil, key: nil)
+    env = Figaro.env
+    key ||= env.attribute_encryption_key
+    cost ||= env.attribute_cost
+    UserAccessKey.new(password: key, salt: key, cost: cost)
+  end
+
+  def self.new_from_decrypted(decrypted, user_access_key = new_user_access_key)
+    encryptor = Pii::PasswordEncryptor.new
+    encrypted = encryptor.encrypt(decrypted, user_access_key)
+    new(encrypted, decrypted: decrypted, user_access_key: user_access_key)
+  end
+
+  def initialize(encrypted, decrypted: nil, cost: nil, user_access_key: nil)
+    self.encrypted = encrypted
```

Comment:
reek complains even if they are inside of `initialize` and only accessed with `attr_reader` ? that is lame!